datalad/local/remove.py

```diff
@@ -0,0 +1,229 @@
+
+    This command relies on the 'drop' command for safe operation. By default,
+    only file content from datasets which will be uninstalled as part of
+    a removal will be dropped. Otherwise file content is retained, such that
+    restoring a previous version also immediately restores file content access,
+    just as it is the case for files directly committed to Git. This default
+    behavior can be changed to always drop content prior removal, for cases
+    where a minimal storage footprint for local datasets installations is
+    desirable.
+
+    Removing a dataset component is always a recursive operation. Removing a
+    directory, removes all content underneath the directory too. If
+    subdatasets are located under a to-be-removed path, they will be
+    uninstalled entirely, and all their content dropped. If any subdataset
+    can not be uninstalled safely, the remove operation will fail and halt.
+    """
+    _params_ = dict(
+        dataset=Parameter(
+            args=("-d", "--dataset"),
+            metavar="DATASET",
+            doc="""specify the dataset to perform remove from.
+            If no dataset is given, the current working directory is used
+            as operation context""",
+            constraints=EnsureDataset() | EnsureNone()),
+        path=Parameter(
+            args=("path",),
+            metavar="PATH",
+            doc="path of a dataset or dataset component to be removed",
+            nargs="*",
+            constraints=EnsureStr() | EnsureNone()),
+        drop=Parameter(
+            args=("--drop",),
+            doc="""which dataset components to drop prior removal. This
+            parameter is passed on to the underlying drop operation as
+            its 'what' argument.""",
+            # we must not offer a 'nothing' which would bypass
+            # the `drop()` call. The implementation completely
+            # relies on `drop()` for all safety measures.
+            # instead `drop(reckless=kill)` must be used to fast-kill
+            # things
+            constraints=EnsureChoice('datasets', 'all')),
+        jobs=jobs_opt,
+        message=save_message_opt,
+        # XXX deprecate!
+        save=Parameter(
+            args=("--nosave",),
+            dest='save',
+            action="store_false",
+            doc="""DEPRECATED and IGNORED; use `save --amend` instead"""),
+        recursive=Parameter(
+            args=("--recursive", '-r',),
+            doc="""DEPRECATED and IGNORED: removal is always a recursive
+            operation"""),
+    )
+    # inherit some from Drop
+    # if_dirty and check as deprecated
+    for p in ('reckless', 'if_dirty', 'check',):
+        _params_[p] = Drop._params_[p]
+
+    _examples_ = [
+        dict(text="Permanently remove a subdataset from a dataset",
+             code_py="remove(dataset='path/to/dataset', path='path/to/subds')",
+             code_cmd="datalad remove -d <path/to/dataset> <path/to/subds>"),
+    ]
+
```

Comment:
maybe emphasize the inherent recursiveness of operations in the examples? And maybe also mention removal of a superdataset as an example (it is my most used usecase for remove)

```suggestion
    _examples_ = [
        dict(text="Permanently remove a subdataset (and all further subdatasets contained in it) from a dataset",
             code_py="remove(dataset='path/to/dataset', path='path/to/subds')",
             code_cmd="datalad remove -d <path/to/dataset> <path/to/subds>"),
             dict(text="Permanently remove a superdataset (with all subdatasets) from the filesystem",
             code_py="remove(dataset='path/to/dataset')",
             code_cmd="datalad remove -d <path/to/dataset>"),
    ]

```

Reply:
Makes sense. Thx!